workflow/controller/workflowpod.go

```diff
@@ -221,6 +222,39 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		}
 	}
 
+	// Apply the patch string from template
+	if woc.wf.Spec.PodSpecPatch != "" || tmpl.PodSpecPatch != "" {
+
+		if tmpl.PodSpecPatch == "" {
+			tmpl.PodSpecPatch = woc.wf.Spec.PodSpecPatch
+			pod, err = substitutePodParams(pod, woc.globalParams, tmpl)
+			if err != nil {
+				return nil, err
+			}
+		}
+		jsonstr, err := json.Marshal(pod.Spec)
```

Comment:
We need to allow pod spec patches to exist in YAML form. 

If we want to support both, we can peek at the first non-whitespace character of the patch string, and if it is `{` we use json unmarshalling. Otherwise yaml unmarshalling.